org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/XMLTextDocumentService.java

```diff
@@ -527,4 +550,29 @@ public boolean documentIsOpen(String uri) {
 		start.complete(cancelIndicator);
 		return result;
 	}
+
+	/**
+	 * Sends a notification that informs the user that the symbols limit has been exceeded
+	 * while computing symbols for textDocument/documentSymbol
+	 * 
+	 * @param xmlDocument    the xml document that symbols were being computed for
+	 * @param symbolSettings the symbol settings
+	 */
+	private void sendSymbolLimitNotification(DOMDocument xmlDocument, XMLSymbolSettings symbolSettings) {

```

Comment:
We should keep track whether the message has already been sent for this URI, for the rest of this JVM session, i.e keep a static Set of impacted URIs. See https://github.com/redhat-developer/vscode-xml/issues/243